tests/components/homematicip_cloud/test_init.py

```diff
@@ -244,3 +245,61 @@ async def test_setup_two_haps_unload_one_by_one(hass: HomeAssistant) -> None:
 
     # Check services are removed
     assert not hass.services.async_services().get(DOMAIN)
+
+
+async def test_remove_obsolete_entities(hass: HomeAssistant) -> None:
+    """Test removing obsolete entities."""
+    mock_config = {HMIPC_AUTHTOKEN: "123", HMIPC_HAPID: "ABC123", HMIPC_NAME: "name"}
+    entry = MockConfigEntry(domain=DOMAIN, data=mock_config)
+    entry.add_to_hass(hass)
+
+    # Create mock entities for all three removal conditions
+    mock_entries = [
+        Mock(entity_id="sensor.ap", unique_id="HomematicipAccesspointStatus123"),
+        Mock(entity_id="sensor.battery", unique_id="HomematicipBatterySensor_ABC123"),
+        Mock(entity_id="sensor.keep", unique_id="ShouldRemain123"),
+    ]
```

Comment:
Don't create entities like this. Rather use the entity_registry normally and use entity_registry.async_get_or_create and create them manually. Trying to mock out the entity registry sounds like a breaking CI test waiting to happen

Reply:
fixed thanks for input